dags/analyzer_helper/discord/discord_transform_raw_data.py

```diff
@@ -0,0 +1,59 @@
+from datetime import datetime
+
+from dags.analyzer_helper.discord.transform_raw_data_base import TransformRawDataBase
+
+
+class DiscordTransformRawData(TransformRawDataBase):
+    def transform(self, raw_data: list, platform_id: str, period: datetime) -> list[dict]:
+        transformed_data = []
+        for data in raw_data:
+            interactions = []
+            if data.get("replied"):
+                interactions.append({
+                    "name": "reply",
+                    "users_engaged_id": [data.get("replied")],
+                    "type": "receiver"
+                })
+
+            if data.get("user_mentions"):
+                interactions.append({
+                    "name": "mention",
+                    "users_engaged_id": data.get("user_mentions"),
+                    "type": "emitter"
+                })
+
+            if data.get("reactions"):
+                for reaction in data.get("reactions"):
+                    parts = reaction.split(", ")
+                    if len(parts) > 2:
+                        interactions.append({
+                            "name": "reaction",
+                            "users_engaged_id": parts[:-1],
+                            "type": "emitter"
+                        })
```

Comment:
Please use an extend to first combining all users engaged and then add it to interactions (after the for loop). Also, the type of interaction for the user would be `receiver`

We also want to create a document for each user in reaction to have the reaction interaction and have the type of `emitter`